Mopro refactor doesn't work for React Native on iOS

# Problem
In PR https://github.com/zkmopro/mopro/pull/561
It works well for iOS/Android on XCode, Android Studio, and Flutter app
However in iOS simulator (built for simulator) it failed on React Native
It shows the error
```sh

Showing Recent Errors Only
Ld /Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/reactnativeapp.app/reactnativeapp.debug.dylib normal (in target 'reactnativeapp' from project 'reactnativeapp')
    cd /Users/zhengyawen/Documents/GitHub/mopro-example-app/react-native/ios
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -Xlinker -reproducible -target arm64-apple-ios15.1-simulator -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator18.5.sdk -O0 -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphonesimulator -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator -L/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator18.5.sdk/usr/lib/swift -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/DoubleConversion -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/EXConstants -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/Expo -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoAsset -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoBlur -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoFileSystem -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoFont -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoHaptics -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoHead -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoKeepAwake -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoLinking -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoModulesCore -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoSplashScreen -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoSymbols -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoSystemUI -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ExpoWebBrowser -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/Mopro -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/RCT-Folly -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/RCTDeprecation -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/RCTTypeSafety -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/RNGestureHandler -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/RNReanimated -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/RNScreens -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-Core -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-CoreModules -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-Fabric -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-FabricComponents -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-FabricImage -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-ImageManager -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-Mapbuffer -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-NativeModulesApple -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RCTAnimation -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RCTAppDelegate -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RCTBlob -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RCTFabric -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RCTImage -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RCTLinking -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RCTNetwork -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RCTSettings -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RCTText -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RCTVibration -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RuntimeApple -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RuntimeCore -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-RuntimeHermes -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-cxxreact -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-debug -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-defaultsnativemodule -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-domnativemodule -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-featureflags -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-featureflagsnativemodule -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-graphics -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-hermes -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-idlecallbacksnativemodule -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-jserrorhandler -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-jsi -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-jsiexecutor -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-jsinspector -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-logger -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-microtasksnativemodule -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-nativeconfig -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-perflogger -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-performancetimeline -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-rendererconsistency -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-rendererdebug -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-runtimescheduler -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/React-utils -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ReactCodegen -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/ReactCommon -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/SocketRocket -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/Yoga -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/fmt -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/glog -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/react-native-safe-area-context -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/react-native-webview -L/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/XCFrameworkIntermediates/Mopro -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator18.5.sdk/usr/lib/swift -F/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphonesimulator -F/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator -F/Users/zhengyawen/Documents/GitHub/mopro-example-app/react-native/modules/mopro/ios/MoproiOSBindings -F/Users/zhengyawen/Documents/GitHub/mopro-example-app/react-native/ios/Pods/hermes-engine/destroot/Library/Frameworks/universal -F/Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/XCFrameworkIntermediates/hermes-engine/Pre-built -filelist /Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Intermediates.noindex/reactnativeapp.build/Debug-iphonesimulator/reactnativeapp.build/Objects-normal/arm64/reactnativeapp.LinkFileList -install_name @rpath/reactnativeapp.debug.dylib -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Intermediates.noindex/reactnativeapp.build/Debug-iphonesimulator/reactnativeapp.build/Objects-normal/arm64/reactnativeapp_lto.o -rdynamic -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -Xlinker -dependency_info -Xlinker /Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Intermediates.noindex/reactnativeapp.build/Debug-iphonesimulator/reactnativeapp.build/Objects-normal/arm64/reactnativeapp_dependency_info.dat -stdlib\=libc++ -fobjc-arc -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Intermediates.noindex/reactnativeapp.build/Debug-iphonesimulator/reactnativeapp.build/Objects-normal/arm64/reactnativeapp.swiftmodule -ObjC -lDoubleConversion -lEXConstants -lExpo -lExpoAsset -lExpoBlur -lExpoFileSystem -lExpoFont -lExpoHaptics -lExpoHead -lExpoKeepAwake -lExpoLinking -lExpoModulesCore -lExpoSplashScreen -lExpoSymbols -lExpoSystemUI -lExpoWebBrowser -lMopro -lRCT-Folly -lRCTDeprecation -lRCTTypeSafety -lRNGestureHandler -lRNReanimated -lRNScreens -lReact-Core -lReact-CoreModules -lReact-Fabric -lReact-FabricComponents -lReact-FabricImage -lReact-ImageManager -lReact-Mapbuffer -lReact-NativeModulesApple -lReact-RCTAnimation -lReact-RCTAppDelegate -lReact-RCTBlob -lReact-RCTFabric -lReact-RCTImage -lReact-RCTLinking -lReact-RCTNetwork -lReact-RCTSettings -lReact-RCTText -lReact-RCTVibration -lReact-RuntimeApple -lReact-RuntimeCore -lReact-RuntimeHermes -lReact-cxxreact -lReact-debug -lReact-defaultsnativemodule -lReact-domnativemodule -lReact-featureflags -lReact-featureflagsnativemodule -lReact-graphics -lReact-hermes -lReact-idlecallbacksnativemodule -lReact-jserrorhandler -lReact-jsi -lReact-jsiexecutor -lReact-jsinspector -lReact-logger -lReact-microtasksnativemodule -lReact-nativeconfig -lReact-perflogger -lReact-performancetimeline -lReact-rendererconsistency -lReact-rendererdebug -lReact-runtimescheduler -lReact-utils -lReactCodegen -lReactCommon -lSocketRocket -lYoga -lc++ -lc++abi -lfmt -lglog -licucore -lmopro -lreact-native-safe-area-context -lreact-native-webview -framework Accelerate -framework AudioToolbox -framework CFNetwork -framework CoreGraphics -framework ImageIO -framework JavaScriptCore -framework MobileCoreServices -framework QuartzCore -framework Security -framework UIKit -framework hermes -ObjC -lc++ -Xlinker -alias -Xlinker _main -Xlinker ___debug_main_executable_dylib_entry_point -lPods-reactnativeapp -Xlinker -no_adhoc_codesign -o /Users/zhengyawen/Library/Developer/Xcode/DerivedData/reactnativeapp-gklqgdgvvpyzbpgleppsyexgdujm/Build/Products/Debug-iphonesimulator/reactnativeapp.app/reactnativeapp.debug.dylib

ld: warning: ignoring duplicate libraries: '-lc++'
ld: warning: Could not find or use auto-linked framework 'CoreAudioTypes': framework 'CoreAudioTypes' not found
ld: warning: Could not parse or use implicit file '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk/System/Library/Frameworks/SwiftUICore.framework/SwiftUICore.tbd': cannot link directly with 'SwiftUICore' because product being built is not an allowed client of it
Undefined symbols for architecture arm64:
  "_ffi_mopro_rustbuffer_free", referenced from:
      closure #1 (Swift.UnsafeMutablePointer<__C.RustCallStatus>) -> () in (extension in Mopro):__C.RustBuffer.(deallocate in _1A658132631AB3C8F7CB5CCDC546BD8F)() -> () in libMopro.a[3](mopro.o)
  "_ffi_mopro_rustbuffer_from_bytes", referenced from:
      closure #1 (Swift.UnsafeMutablePointer<__C.RustCallStatus>) -> __C.RustBuffer in static (extension in Mopro):__C.RustBuffer.(from in _1A658132631AB3C8F7CB5CCDC546BD8F)(Swift.UnsafeBufferPointer<Swift.UInt8>) -> __C.RustBuffer in libMopro.a[3](mopro.o)
  "_ffi_mopro_uniffi_contract_version", referenced from:
      closure #1 () -> Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in variable initialization expression of Mopro.(initializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) : Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in libMopro.a[3](mopro.o)
  "_uniffi_mopro_checksum_func_generate_circom_proof", referenced from:
      closure #1 () -> Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in variable initialization expression of Mopro.(initializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) : Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in libMopro.a[3](mopro.o)
  "_uniffi_mopro_checksum_func_generate_halo2_proof", referenced from:
      closure #1 () -> Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in variable initialization expression of Mopro.(initializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) : Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in libMopro.a[3](mopro.o)
  "_uniffi_mopro_checksum_func_generate_noir_proof", referenced from:
      closure #1 () -> Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in variable initialization expression of Mopro.(initializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) : Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in libMopro.a[3](mopro.o)
  "_uniffi_mopro_checksum_func_get_noir_verification_key", referenced from:
      closure #1 () -> Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in variable initialization expression of Mopro.(initializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) : Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in libMopro.a[3](mopro.o)
  "_uniffi_mopro_checksum_func_mopro_uniffi_hello_world", referenced from:
      closure #1 () -> Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in variable initialization expression of Mopro.(initializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) : Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in libMopro.a[3](mopro.o)
  "_uniffi_mopro_checksum_func_verify_circom_proof", referenced from:
      closure #1 () -> Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in variable initialization expression of Mopro.(initializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) : Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in libMopro.a[3](mopro.o)
  "_uniffi_mopro_checksum_func_verify_halo2_proof", referenced from:
      closure #1 () -> Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in variable initialization expression of Mopro.(initializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) : Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in libMopro.a[3](mopro.o)
  "_uniffi_mopro_checksum_func_verify_noir_proof", referenced from:
      closure #1 () -> Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in variable initialization expression of Mopro.(initializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) : Mopro.(InitializationResult in _1A658132631AB3C8F7CB5CCDC546BD8F) in libMopro.a[3](mopro.o)
  "_uniffi_mopro_fn_func_generate_circom_proof", referenced from:
      closure #1 (Swift.UnsafeMutablePointer<__C.RustCallStatus>) -> __C.RustBuffer in Mopro.generateCircomProof(zkeyPath: Swift.String, circuitInputs: Swift.String, proofLib: Mopro.ProofLib) throws -> Mopro.CircomProofResult in libMopro.a[3](mopro.o)
  "_uniffi_mopro_fn_func_generate_halo2_proof", referenced from:
      closure #1 (Swift.UnsafeMutablePointer<__C.RustCallStatus>) -> __C.RustBuffer in Mopro.generateHalo2Proof(srsPath: Swift.String, pkPath: Swift.String, circuitInputs: [Swift.String : [Swift.String]]) throws -> Mopro.Halo2ProofResult in libMopro.a[3](mopro.o)
  "_uniffi_mopro_fn_func_generate_noir_proof", referenced from:
      closure #1 (Swift.UnsafeMutablePointer<__C.RustCallStatus>) -> __C.RustBuffer in Mopro.generateNoirProof(circuitPath: Swift.String, srsPath: Swift.String?, inputs: [Swift.String], onChain: Swift.Bool, vk: Foundation.Data, lowMemoryMode: Swift.Bool) throws -> Foundation.Data in libMopro.a[3](mopro.o)
  "_uniffi_mopro_fn_func_get_noir_verification_key", referenced from:
      closure #1 (Swift.UnsafeMutablePointer<__C.RustCallStatus>) -> __C.RustBuffer in Mopro.getNoirVerificationKey(circuitPath: Swift.String, srsPath: Swift.String?, onChain: Swift.Bool, lowMemoryMode: Swift.Bool) throws -> Foundation.Data in libMopro.a[3](mopro.o)
  "_uniffi_mopro_fn_func_mopro_uniffi_hello_world", referenced from:
      closure #1 (Swift.UnsafeMutablePointer<__C.RustCallStatus>) -> __C.RustBuffer in Mopro.moproUniffiHelloWorld() -> Swift.String in libMopro.a[3](mopro.o)
  "_uniffi_mopro_fn_func_verify_circom_proof", referenced from:
      closure #1 (Swift.UnsafeMutablePointer<__C.RustCallStatus>) -> Swift.Int8 in Mopro.verifyCircomProof(zkeyPath: Swift.String, proofResult: Mopro.CircomProofResult, proofLib: Mopro.ProofLib) throws -> Swift.Bool in libMopro.a[3](mopro.o)
  "_uniffi_mopro_fn_func_verify_halo2_proof", referenced from:
      closure #1 (Swift.UnsafeMutablePointer<__C.RustCallStatus>) -> Swift.Int8 in Mopro.verifyHalo2Proof(srsPath: Swift.String, vkPath: Swift.String, proof: Foundation.Data, publicInput: Foundation.Data) throws -> Swift.Bool in libMopro.a[3](mopro.o)
  "_uniffi_mopro_fn_func_verify_noir_proof", referenced from:
      closure #1 (Swift.UnsafeMutablePointer<__C.RustCallStatus>) -> Swift.Int8 in Mopro.verifyNoirProof(circuitPath: Swift.String, proof: Foundation.Data, onChain: Swift.Bool, vk: Foundation.Data, lowMemoryMode: Swift.Bool) throws -> Swift.Bool in libMopro.a[3](mopro.o)
ld: symbol(s) not found for architecture arm64
clang++: error: linker command failed with exit code 1 (use -v to see invocation)

Undefined symbol: _ffi_mopro_rustbuffer_free

Undefined symbol: _ffi_mopro_rustbuffer_from_bytes

Undefined symbol: _ffi_mopro_uniffi_contract_version

Undefined symbol: _uniffi_mopro_checksum_func_generate_circom_proof

Undefined symbol: _uniffi_mopro_checksum_func_generate_halo2_proof

Undefined symbol: _uniffi_mopro_checksum_func_generate_noir_proof

Undefined symbol: _uniffi_mopro_checksum_func_get_noir_verification_key

Undefined symbol: _uniffi_mopro_checksum_func_mopro_uniffi_hello_world

Undefined symbol: _uniffi_mopro_checksum_func_verify_circom_proof

Undefined symbol: _uniffi_mopro_checksum_func_verify_halo2_proof

Undefined symbol: _uniffi_mopro_checksum_func_verify_noir_proof

Undefined symbol: _uniffi_mopro_fn_func_generate_circom_proof

Undefined symbol: _uniffi_mopro_fn_func_generate_halo2_proof

Undefined symbol: _uniffi_mopro_fn_func_generate_noir_proof

Undefined symbol: _uniffi_mopro_fn_func_get_noir_verification_key

Undefined symbol: _uniffi_mopro_fn_func_mopro_uniffi_hello_world

Undefined symbol: _uniffi_mopro_fn_func_verify_circom_proof

Undefined symbol: _uniffi_mopro_fn_func_verify_halo2_proof

Undefined symbol: _uniffi_mopro_fn_func_verify_noir_proof


```

## Details
- Make sure the React Native template/CLI works

## Acceptance criteria
- Make sure the React Native template/CLI works
- (optional) add tests for React Native

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -382,6 +382,107 @@ jobs:
             name: android-test-report
             path: ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}/android/app/build/reports/androidTests/connected/debug/
 
+  cli_create_react_native:
+    needs: cli_build
+    strategy:
+      fail-fast: false
+      matrix:
+        adapter: [circom, halo2, noir]
+    runs-on: macos-14
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo_full_name
+    steps:
+      - uses: actions/checkout@v5
+
+      # Build mopro example project for iOS + Android first, and create RN frameworks
+      - name: Prepare template project
+        uses: ./.github/actions/mopro-init-project
+        with:
+          artifact-name: mopro-bin-macOS
+          adapter: ${{ matrix.adapter }}
+          project-name: mopro-example-${{ matrix.adapter }}
+          destination: ${{ runner.temp }}
+
+      - name: Build iOS (debug; device+sim arm64 only)
+        working-directory: ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}
+        run: |
+          "$GITHUB_WORKSPACE/bin/mopro" build \
+            --mode debug \
+            --platforms ios \
+            --architectures aarch64-apple-ios aarch64-apple-ios-sim
+
+      - name: Build Android (debug, x86_64)
+        working-directory: ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}
+        run: |
+          "$GITHUB_WORKSPACE/bin/mopro" build \
+            --mode debug \
+            --platforms android \
+            --architectures x86_64-linux-android
+
+      - name: Clean mopro build dir
+        run: rm -rf ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}/build
+
+      - name: Create React Native frameworks
+        working-directory: ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}
+        run: $GITHUB_WORKSPACE/bin/mopro create --framework react-native
+
+      # ---- Node & deps ----
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Install JS packages
+        working-directory: ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}/react-native
+        run: npm ci
+
+      - name: Expo prebuild (generate ios/ & android/)
+        working-directory: ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}/react-native
+        env:
+          EXPO_NO_INTERACTIVE: '1'
+        run: npx expo prebuild --clean
+
+      # ---- iOS build (headless) ----
+      - name: Install iOS Pods
+        working-directory: ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}/react-native/ios
+        run: pod install
+
+      - name: Build iOS (Debug, iphonesimulator)
+        working-directory: ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}/react-native
+        run: |
+          xcodebuild \
+            -workspace ios/*.xcworkspace \
+            -scheme reactnativeapp \
+            -configuration Debug \
+            -sdk iphonesimulator \
+            ARCHS=arm64 EXCLUDED_ARCHS='x86_64' ONLY_ACTIVE_ARCH=YES \
+            CODE_SIGNING_ALLOWED=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY="" \
+            build
+
+      # ---- Android build (headless) ----
+      - name: Setup Java 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: '17'
+          cache: 'gradle'
+
+      - name: Android SDK (cmdline only)
+        uses: android-actions/setup-android@v3
+
+      - name: Assemble Android Debug (no emulator)
+        working-directory: ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}/react-native/android
+        run: ./gradlew --no-daemon --stacktrace assembleDebug
+
+      - name: Upload native build outputs (on failure)
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: rn-build-outputs-${{ matrix.adapter }}
+          path: |
+            ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}/react-native/ios/build/**/*.log
+            ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}/react-native/android/app/build/outputs/**
+          if-no-files-found: ignore
+
   cli_build_web:
       needs: cli_build
       strategy:

--- a/cli/src/init.rs
+++ b/cli/src/init.rs
@@ -71,6 +71,10 @@ pub fn init_project(
         adapter_sel.lib_template(lib_rs_path);
     }
 
+    if let Some(test_bindings_dir_path) = project_dir.join("tests").join("bindings").to_str() {
+        replace_test_bindings_lib_import(test_bindings_dir_path, project_name.as_str())?;
+    }
+
     // Store selection
     let config_path = project_dir.join("Config.toml");
 
@@ -93,6 +97,50 @@ pub fn init_project(
     if !quiet {
         // Print out the instructions
         print_init_instructions(project_name);
+    }
+
+    Ok(())
+}
+
+/// Replace the placeholder import lines in the test bindings files with the actual import statements
+/// based on the project name and file type (Kotlin or Swift).
+fn replace_test_bindings_lib_import(test_bindings_dir: &str, project_name: &str) -> Result<()> {
+    let project_lib_name = project_name.replace('-', "_");
+
+    // We expect a structure of `tests/bindings/<adapter>/*.(kts|swift)`
+    for bundle in fs::read_dir(test_bindings_dir)? {
+        let bundle = bundle?;
+        let bundle_path = bundle.path();
+
+        if !bundle_path.is_dir() {
+            continue;
+        }
+
+        for file in fs::read_dir(&bundle_path)? {
+            let file = file?;
+            let file_path = file.path();
+
+            if !file_path.is_file() {
+                continue;
+            }
+
+            let target = "// GENERATED LIB IMPORT PLACEHOLDER";
+            let replacement = match file_path.extension().and_then(|ext| ext.to_str()) {
+                Some("kts") => Some(format!("import uniffi.{}.*", project_lib_name)),
+                Some("swift") => Some(format!("import {}", project_lib_name)),
+                _ => None,
+            };
+
+            if let Some(line) = replacement {
+                replace_string_in_file(
+                    file_path
+                        .to_str()
+                        .ok_or(anyhow::anyhow!("Invalid test binding file path"))?,
+                    target,
+                    &line,
+                )?;
+            }
+        }
     }
 
     Ok(())

--- a/cli/src/init/write_toml.rs
+++ b/cli/src/init/write_toml.rs
@@ -6,7 +6,6 @@ version = "0.1.0"
 edition = "2021"
 
 [lib]
-name = "mopro"
 crate-type = ["lib", "cdylib", "staticlib"]
 
 # Adapters for different proof systems

--- a/cli/src/template/init/tests/bindings/circom/test_multiplier2.kts
+++ b/cli/src/template/init/tests/bindings/circom/test_multiplier2.kts
@@ -1,4 +1,4 @@
-import uniffi.mopro.*
+// GENERATED LIB IMPORT PLACEHOLDER
 
 try {
     var zkeyPath = "./test-vectors/circom/multiplier2_final.zkey"

--- a/cli/src/template/init/tests/bindings/circom/test_multiplier2.swift
+++ b/cli/src/template/init/tests/bindings/circom/test_multiplier2.swift
@@ -1,5 +1,5 @@
 import Foundation
-import mopro
+// GENERATED LIB IMPORT PLACEHOLDER
 
 do {
     let zkeyPath = "../../../test-vectors/circom/multiplier2_final.zkey"

--- a/cli/src/template/init/tests/bindings/halo2/test_gemini_fibonacci.kts
+++ b/cli/src/template/init/tests/bindings/halo2/test_gemini_fibonacci.kts
@@ -1,4 +1,4 @@
-import uniffi.mopro.*
+// GENERATED LIB IMPORT PLACEHOLDER
 
 try {
     val srsPath = "./test-vectors/halo2/gemini_fibonacci_srs.bin"

--- a/cli/src/template/init/tests/bindings/halo2/test_gemini_fibonacci.swift
+++ b/cli/src/template/init/tests/bindings/halo2/test_gemini_fibonacci.swift
@@ -1,5 +1,5 @@
 import Foundation
-import mopro
+// GENERATED LIB IMPORT PLACEHOLDER
 
 do {
     let srsPath = "../../../test-vectors/halo2/gemini_fibonacci_srs.bin"

--- a/cli/src/template/init/tests/bindings/halo2/test_hyperplonk_fibonacci.kts
+++ b/cli/src/template/init/tests/bindings/halo2/test_hyperplonk_fibonacci.kts
@@ -1,4 +1,4 @@
-import uniffi.mopro.*
+// GENERATED LIB IMPORT PLACEHOLDER
 
 try {
     val srsPath = "./test-vectors/halo2/hyperplonk_fibonacci_srs.bin"

--- a/cli/src/template/init/tests/bindings/halo2/test_hyperplonk_fibonacci.swift
+++ b/cli/src/template/init/tests/bindings/halo2/test_hyperplonk_fibonacci.swift
@@ -1,5 +1,5 @@
 import Foundation
-import mopro
+// GENERATED LIB IMPORT PLACEHOLDER
 
 do {
     let srsPath = "../../../test-vectors/halo2/hyperplonk_fibonacci_srs.bin"

--- a/cli/src/template/init/tests/bindings/halo2/test_plonk_fibonacci.kts
+++ b/cli/src/template/init/tests/bindings/halo2/test_plonk_fibonacci.kts
@@ -1,4 +1,4 @@
-import uniffi.mopro.*
+// GENERATED LIB IMPORT PLACEHOLDER
 
 try {
     val srsPath = "./test-vectors/halo2/plonk_fibonacci_srs.bin"

--- a/cli/src/template/init/tests/bindings/halo2/test_plonk_fibonacci.swift
+++ b/cli/src/template/init/tests/bindings/halo2/test_plonk_fibonacci.swift
@@ -1,5 +1,5 @@
 import Foundation
-import mopro
+// GENERATED LIB IMPORT PLACEHOLDER
 
 do {
     let srsPath = "../../../test-vectors/halo2/plonk_fibonacci_srs.bin"

